src/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexFCD.cs

```diff
@@ -87,7 +87,12 @@ internal static RegexPrefix Prefix(RegexTree tree)
 
                     case RegexNode.Oneloop:
                     case RegexNode.Onelazy:
-                        if (curNode._m > 0)
+
+                        int cutoff = 1_000_000; // In release, cutoff at a length to which we can reasonably construct a string
+#if DEBUG
+                        cutoff = 50; // In debug, use a smaller cutoff to exercise the cutoff path
```

Comment:
Note: this is fairly small.  This would mean a totally reasonably regex of "a{100,110}" would fail.  I don't know what contexts debug builds are used in.  but something to be potentially worried about.  Perhaps another way would be to have a static and allow tests to change it.

Reply:
Hopefully not fail - this just disables the prefix optimization for >50.

Reply:
Re static - we try to avoid more use of internalsvisibleto